### PR TITLE
fix: avoid merge conflicts during properties collection

### LINKED ISSUE #11404: BeanPropertySet.get(...) fails on types that inherit the same interface multiple times
### Description of the bug
BeanPropertySet.get(...) fails on types that inherit the same interface multiple times.

### Minimal reproducible example
```
  interface Iface3
    extends
      Iface2,
      Iface
  {
  }

  interface Iface2
    extends
      Iface
  {
  }

  interface Iface
  {
    Object getPropertyValue();
  }

  public static void main(String[] args)
    throws IntrospectionException
  {
    // lists the property twice
    BeanUtil.getBeanPropertyDescriptors(Iface3.class).stream().forEach(property -> {
      System.out.println(property.getName());
    });

    // fails because of the duplicate property
    BeanPropertySet.get(Iface3.class, false, PropertyFilterDefinition.getDefaultFilter());
  }

```

### Expected behavior
Bean properties declared multiple times in the hierarchy shouldn't cause problems.
### Actual behavior
Exception is thrown:
` java.lang.IllegalStateException: Duplicate key com.vaadin.flow.data.binder.BeanPropertySet$BeanPropertyDefinition@64cd705f`

### Versions:
    - Vaadin / Flow version: Vaadin 20
    - Java version: tested with 8 and 11
    - OS version: Windows

### PR DESCRIPTION
**Pull request by denis-anisimov**
fixes #11404

## Description

Please list all relevant dependencies in this section and provide summary of the change, motivation and context.

Fixes # (issue)

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-data/src/main/java/com/vaadin/flow/data/binder/BeanPropertySet.java [hunk 1]
```diff
@@ -323,7 +323,8 @@ private BeanPropertySet(InstanceKey<T> instanceKey) {
                     .map(descriptor -> new BeanPropertyDefinition<>(this,
                             instanceKey.type, descriptor))
                     .collect(Collectors.toMap(PropertyDefinition::getName,
-                            Function.identity()));
+                            Function.identity(), this::mergePropertyDefinitions,
+                            HashMap::new));
```
Comment (mshabarov) [inline flow-data/src/main/java/com/vaadin/flow/data/binder/BeanPropertySet.java:327]: Why do you specify `HashMap::new` here and not use overloaded method where it is used by default?

**Review by mshabarov — DISMISSED** [review]

File: flow-data/src/main/java/com/vaadin/flow/data/binder/BeanPropertySet.java (hunk 1 above)
Comment (denis-anisimov) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/BeanPropertySet.java:327]: Apparently I missed the method which doesn't require the last argument.

Comment (denis-anisimov) [reply flow-data/src/main/java/com/vaadin/flow/data/binder/BeanPropertySet.java]: done

**Review by mshabarov — APPROVED** [review]

**Comment by mshabarov** [comment]
@denis-anisimov I'm not sure is it an inconsistency, but if `BeanPropertySet.get` returns only one property definition for the considered example, then I would expect `BeanUtil.getBeanPropertyDescriptors` also returns only one descriptor for inherited beans.

**Comment by denis-anisimov** [comment]
`BeanUtil` is a totally different util class in a different module.
If it has an issue then this is a totally different issue and it doesn't cause any exception.
The PR should address only one issue, not both.

**Pull request merged**